packages/device-api/readme.md

```diff
@@ -0,0 +1,299 @@
+
+**`validators.zod.js` example**
+
+```javascript
+import z from "zod";
+
+export const getAliasResultValidator = z.object({
+    success: z.string()
+})
+```
+
+**`deviceApiCalls.js` example**
+```javascript
+import { getAliasResultValidator } from "./validators.zod.js"
+
+/** @extends {DeviceApiCall<getAliasResultValidator>} */
+export class GetAlias extends DeviceApiCall {
+    method = "getAlias"
+    resultValidator = getAliasResultValidator
+}
+```
+
+The most important part of this setup, is that `import z from "zod"` only occurs from within `.zod.js` files.
+
+## Working with DeviceApiTransport's
+
+### 1) Implement a new transport
+
+A `DeviceApiTransport` handles the very last part of browser<->native messaging. The following example is
+a cut-down version to show the concept. 
+
+Implementors just override `send(deviceApiCall)` and can do whatever is needed on their platform to respond.
+
+```javascript
+// an example transport 
+class AppleTransport extends DeviceApiTransport {
+    async send (deviceApiCall) {
+        return wkSendAndWait(deviceApiCall.method, deviceApiCall.params, {
+            secret: this.config.secret,
+            hasModernWebkitAPI: this.config.hasModernWebkitAPI
+        })
+    }
+}
+```
+
+If this DeviceApiTransport is chosen for the current platform, then every call to `this.deviceApi.request` and `this.deviceApi.notify`
+will come through here.
+
+### 2) Debug an existing transport 
+
+Search for anything that `extends DeviceApiTransport` -> there's only a single method implemented, so it should be
+straightforward to incorporate logging to see what's going on.
+
+### 3) Make a change to an existing transport 
+
+Once you know of a required change, you may need to update supporting mocks that live within `integration-test/helpers`
+
+---
+
+## Internals
+
+This library contains:
+
+- `DeviceApiCall` - a class that you can extend to describe your device api calls, with validation via `Zod`
+- `DeviceApiTransport` - each platform can implement to customise how requests & notifications work
+    - Example implementation: [Apple](../../src/appleDeviceUtils/appleDeviceUtils.js#L97-L115)  
+- `DeviceApi` - the API that your device interfaces will use to make requests & notifications
+
+Note: This is designed to roughly follow naming patterns seen here: https://www.jsonrpc.org/specification 
+Note: This is designed to roughly follow naming patterns seen here: https://www.jsonrpc.org/specification 
+Note: This is designed to roughly follow naming patterns seen here: https://www.jsonrpc.org/specification 
+
+## `DeviceApiCall`
+
+**Class format**
+
+You should use this format for all *new* device api calls
+
+Benefits:
+  1) By using a `class` for each new message, we enable powerful type checking
+     1) For example, if a given transport want to intercept or change an outgoing api call, it can use `instanceof` to get full type safety on parameters and result types. 
```

Comment:
```suggestion
     1) For example, if a given transport wants to intercept or change an outgoing api call, it can use `instanceof` to get full type safety on parameters and result types. 
```